test/test_dhcpv6.py

```diff
@@ -27,6 +27,16 @@ def test_dhcpv6(prepare_ifaces):
 	assert answer[DHCP6OptIA_NA].iaid == IAID, \
 		f"Bad IA id in DHCPv6 Solicit"
 
+	# Extracting the DNS servers from the DHCPv6 reply
+	dns_servers = None
+	if DHCP6OptDNSServers in answer:
+		dns_servers = answer[DHCP6OptDNSServers].dnsservers
+	else:
+		raise AssertionError("No DNS servers option in DHCPv6 reply")
+
+	if not dns_servers or dhcpv6_dns1 not in dns_servers or dhcpv6_dns2 not in dns_servers:
+		raise AssertionError(f"DHCPv6 reply does not specify the correct DNS servers: {dns_servers} instead of {dhcpv6_dns1} and {dhcpv6_dns2}")
+
```

Comment:
Can you change this to use `assert DHCP6OptDNSServers in answer, "No DNS...` and similar for the second condition too?

For one thing, it's more consistent with the rest of the code, for other, AssertionError is an exception, whereas asserts can be actually turned off globally by `-O`

Reply:
This exact same DNS check is actually handled with AssertionError for dhcpv4 case so this was an attempt to stay consistent with the current code base. Will change to asserts. Let's see what we can do with the dhcpv4 AssertionError exceptions later on.